packages/report-server/src/reportBuilder/transform/aliases/summaryAliases.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { Row } from '../../types';
+
+/**
+ * Inserts a new column with a value that is the calculation of the percentage of 1s out of 1s and 0s in a row.
+ * { rowFieldName: 'Availability of Male Condoms', facilityNameA: 'N', facilityNameB: 'N', facilityNameC: 'Y', facilityNameD: 'N'  }
+ *  => { rowFieldName: 'Availability of Male Condoms', facilityNameA: 'N', facilityNameB: 'N', facilityNameC: 'Y', facilityNameD: 'N', rowSummary: 0.75 }
+ */
+
+const detectColumnsToSummarise = (rows: Row[]): string[] => {
```

Comment:
One neat thing about typescript is that it performs type inference:
```suggestion
const detectColumnsToSummarise = (rows: Row[]) => {
```
You don't need to be explicit that this function returns a `string[]` as typescript will infer that based on the return value of the function. Likewise, later on when doing `const columnsToSummarise = detectColumnsToSummarise(rows);` you don't need to specify that `columnsToSummarise` is a `string[]` as typescript will infer that as well.

The advantage of letting type inference do its magic is that in the scenario that you want to change the interface for something (eg. lets say we want this function to return a `Set` or a `Map`) then we just need to change the return value and everything automatically updates. Ideally, we are only explicit about types in the scenarios that it matters (parameters into functions, or in some scenarios that we want to be explicit for some reason).